src/library_wasmfs_node.js

```diff
@@ -54,39 +64,35 @@ addToLibrary({
   // https://github.com/google/closure-compiler/pull/4093
   _wasmfs_node_readdir__docs: '/** @suppress {checkTypes} */',
   _wasmfs_node_readdir__deps: [
-    '$wasmfsNodeConvertNodeCode',
+    '$wasmfsTry',
     '$stackSave',
     '$stackRestore',
     '$stringToUTF8OnStack',
     '_wasmfs_node_record_dirent',
   ],
   _wasmfs_node_readdir: (path_p, vec) => {
     let path = UTF8ToString(path_p);
-    let entries;
-    try {
-      entries = fs.readdirSync(path, { withFileTypes: true });
-    } catch (e) {
-      if (!e.code) throw e;
-      return wasmfsNodeConvertNodeCode(e);
-    }
-    entries.forEach((entry) => {
-      let sp = stackSave();
-      let name = stringToUTF8OnStack(entry.name);
-      let type;
-      // TODO: Figure out how to use `cDefine` here.
-      if (entry.isFile()) {
-        type = 1;
-      } else if (entry.isDirectory()) {
-        type = 2;
-      } else if (entry.isSymbolicLink()) {
-        type = 3;
-      } else {
-        type = 0;
-      }
-      __wasmfs_node_record_dirent(vec, name, type);
-      stackRestore(sp);
+    return wasmfsTry(() => {
+      let entries = fs.readdirSync(path, { withFileTypes: true });
```

Comment:
Should the try be only on this line? It looks like now it includes the lines after as well.

Reply:
Is that not worth it? In general I think we've preferred such scopes to be as limited as possible, to avoid the risk of unrelated errors being caught.

Reply:
The handler here just rethrow stuff that doesn't have a `code` so it wouldn't be a big issue I think,

Reply:
Fair enough, I don't feel strongly.